File: Python/Algorithms/345.py
class Solution:
    def reverseVowels(self, s: str) -> str:
        start, end = 0, len(s) - 1
        vowels = {*"aeiouAEIOU"}
        l = list(s)
        while start < end:
            while end > start and l[end] not in vowels:
                end -= 1
            while start < end and l[start] not in vowels:
                start += 1

            l[end], l[start] = l[start], l[end]
            start += 1
            end -= 1
        return "".join(l)

File: Javascript/Algorithms/345.js
/**
 * @param {string} s
 * @return {string}
 */
var reverseVowels = function(s) {
  let start = 0, end = s.length - 1;
  const vowels = new Set(...['aeiouAEIOU']), l = [...s];
  while (start < end) {
    while (end > start && !vowels.has(l[end])) end--;
    while (end > start && !vowels.has(l[start])) start++;

    let tmp = l[end];
    l[end--] = l[start];
    l[start++] = tmp;
  }
  return l.join('');
};